pkg/frontend/querymiddleware/astmapper/instant_splitting.go

```diff
@@ -194,10 +196,16 @@ func (i *instantSplitter) mapCall(expr *parser.Call) (mapped parser.Expr, finish
 		return i.mapCallAvgOverTime(expr)
 	case countOverTime:
 		return i.mapCallVectorAggregation(expr, parser.SUM)
+	case increase:
+		return i.mapCallVectorAggregation(expr, parser.SUM)
+	case irate:
+		return i.mapCallRate(expr)
```

Comment:
I don't think it's correct. Implementing it this way, we run `rate()` and `irate()` the same way with splitting, but `irate()` is different from `rate()`. `irate()` only takes in account the last 2 samples of the range vector.

A rough idea I have to split `irate()` is, for each series, to take the sample computed by the most recent partial/split query, but I'm not sure it's worth the effort. I would rather skip `irate()` for now.

Reply:
Thanks for the clarification. I misunderstood the result of `irate`. Will remove it for now.

> A rough idea I have to split irate() is, for each series, to take the sample computed by the most recent partial/split query, but I'm not sure it's worth the effort. I would rather skip irate() for now.

But we cannot rely only on the most recent partial query, right? If it takes into account the last 2 sample values we cannot know for sure that these last 2 samples are in the last split query, they could be distributed across different partial queries 🤔

Reply:
>But we cannot rely only on the most recent partial query, right?

We can't because of the reason you mentioned. There's no guarantee the last 2 samples of a series are within the last split (it could be a stale series and last 2 samples are in one of the previous splits).